ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -1662,6 +1664,15 @@ static ByteBufferInputStream getByteBufferInputStreamForBlobRecord(CrcInputStrea
       byteBuffer.limit(oldLimit);
       // Change the byte buffer's position as if the data is fetched.
       byteBuffer.position(startIndex + (int) dataSize);
+    } else if (inputStream instanceof NettyByteBufDataInputStream ) {
```

Comment:
Could you add a check to this method to ensure that `dataSize` is within the int range? alternatively, maybe we change the dataSize parameter to an int since every part of this method requires it to be so.